File: src/patterns/structural/facade/facade.go
package main

import (
	"fmt"
	"net/http"
)

type IHttpFacade interface {
	GetAndListen()
}

type HttpFacade struct {
	server *http.Server
	mux *http.ServeMux
}

func NewHttpFacade() *HttpFacade {
	mux := http.NewServeMux()
	server := &http.Server{
		Addr: ":6969",
		Handler:mux,
	}

	return &HttpFacade{mux:mux, server:server}
}

func (h *HttpFacade) GetAndListen() {
	h.mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
		fmt.Fprintf(w, "Hello")
	})

	h.server.ListenAndServe()
}

func main() {
	server := NewHttpFacade()
	server.GetAndListen()
}

File: src/patterns/structural/facade/facade.ts
import express, { Request, Response } from "express";

interface IHttpFacade {
  getAndListen(): void;
}

class HttpFacade implements IHttpFacade {
  app: express.Application;
  constructor(app: express.Application) {
    this.app = app;
  }

  getAndListen() {
    this.app.get("/", (req: Request, res: Response) => {
      res.send("Hello");
    });

    this.app.listen(6969, () => {
      console.log("Listening on port 6969");
    });
  }
}

const server = new HttpFacade(express());
server.getAndListen();
